src/ledger/LedgerManagerImpl.cpp

```diff
@@ -858,6 +873,84 @@ LedgerManagerImpl::setupLedgerCloseMetaStream()
         }
     }
 }
+void
+LedgerManagerImpl::maybeResetLedgerCloseMetaDebugStream(uint32_t ledgerSeq)
+{
+    if (mApp.getConfig().METADATA_DEBUG_LEDGERS != 0)
+    {
+        if (mMetaDebugStream)
+        {
+            if (!FlushAndRotateMetaDebugWork::isDebugSegmentBoundary(ledgerSeq))
+            {
+                // If we've got a stream open and aren't at a reset boundary,
+                // just return -- keep streaming into it.
+                return;
+            }
+
+            // If we have an open stream and there is already a flush-and-rotate
+            // work _running_ (measured by the existence of a shared_ptr at the
+            // end of the weak_ptr), we're in a slightly awkward position since
+            // we can't synchronously close the stream (that would fsync) and we
+            // can't asynchronously close the stream either (that would race
+            // against the running work that is presumably flushing-and-rotating
+            // the _previous_ stream). So we just skip an iteration of launching
+            // flush-and-rotate work, and try again on the next boundary.
+            auto existing = mFlushAndRotateMetaDebugWork.lock();
+            if (existing && !existing->isDone())
+            {
+                CLOG_DEBUG(Ledger,
+                           "Skipping flush-and-rotate of {} since work already "
+                           "running",
+                           mMetaDebugPath.string());
+                return;
+            }
+
+            // If we are resetting and already have a stream, hand it off to the
+            // flush-and-rotate work to finish up with.
+            mFlushAndRotateMetaDebugWork =
+                mApp.getWorkScheduler()
+                    .scheduleWork<FlushAndRotateMetaDebugWork>(
+                        mMetaDebugPath, std::move(mMetaDebugStream),
+                        mApp.getConfig().METADATA_DEBUG_LEDGERS);
+            mMetaDebugPath.clear();
+        }
+
+        // From here on we're starting a new stream, whether it's the first
+        // such stream or a replacement for the one we just handed off to
+        // flush-and-rotate. Either way, we should not have an existing one!
+        releaseAssert(!mMetaDebugStream);
+        mMetaDebugStream = std::make_unique<XDROutputFileStream>(
+            mApp.getClock().getIOContext(),
+            /*fsyncOnClose=*/true);
+
+        mMetaDebugPath = FlushAndRotateMetaDebugWork::getMetaDebugFilePath(
```

Comment:
Because it holds the name of the path being-streamed-to between the time it's opened and the time it's handed off to the close-and-rotate work, 20 minutes later.

Reply:
ah yes indeed, the if right above - this is so that it can gzip it